nucypher/network/nodes.py

```diff
@@ -102,6 +102,11 @@ def __bytes__(self):
     def stamp(self) -> bytes:
         return self.processed_objects['verifying_key'][0]
 
+    @property
+    def domain(self) -> str:
+        domain_bytes = PartiallyKwargifiedBytes.__getattr__(self, "domain")
```

Comment:
Does it make sense to store the result of the first run in a member variable (eg. `self.domain`) so that you don't have to get the bytes and then decode it every time?

Reply:
As an optimization, you mean?

Since we will almost always do this once per node, I think it might actually be a counter-optimization.  The timing here is tiny to begin with: it takes about 1ms to decode 5000 strings on my laptop.  However, it takes 3ms to decode 5000 strings and use the result to set an attribute on an instance.  See this gist: https://gist.github.com/jMyles/e6458de8a557274883e1f2c6cf2e5862

So I think that setting an attribute will actually be slower.  Either way, I think this optimization is premature.  I am definitely interested in ensuring, in the coming weeks, that the total runtime of a grant or retrieve is as slim as possible, but there are places where we can probably shave off hundreds of milliseconds, so I'd prefer to leave single milliseconds alone for the moment until we consider it more holistically.

Note: `PartiallyKwargifiedBytes` already caches its result (which is substantially more onerous) internally.